src/proto/message_translation/tbots_protobuf.h

```diff
@@ -127,23 +128,47 @@ std::unique_ptr<TbotsProto::PlotJugglerValue> createPlotJugglerValue(
     const std::map<std::string, double>& values);
 
 /**
- * Returns a TbotsProto::DebugShapesMap proto containing the name
- * shape pairs.
+ * Returns a TbotsProto::DebugShapes::DebugShape proto for a given shape,
+ * unique id, and an optional debug text.
+ *
+ * @tparam Shape A shape which has a matching createShapeProto function
+ *
+ * @param shape The shape to create the protobuf for
+ * @param unique_id A unique string used to differentiate this shape
+ * from others. Note that this ID should remain the same for the same shape
+ * to avoid it from being plotted multiple times.
+ * @param debug_text An optional string to display on the shape
+ *
+ * @return The unique_ptr to a TbotsProto::DebugShapes::DebugShape proto
+ */
+template <class Shape>
+std::unique_ptr<TbotsProto::DebugShapes::DebugShape> createDebugShape(
+    const Shape& shape, const std::string& unique_id, const std::string& debug_text = "")
+{
+    auto debug_shape = std::make_unique<TbotsProto::DebugShapes::DebugShape>();
+    (*debug_shape->mutable_shape()) = *createShapeProto(shape);
+    debug_shape->set_unique_id(unique_id);
+    debug_shape->set_debug_text(debug_text);
+    return debug_shape;
+};
+
+/**
+ * Returns a TbotsProto::DebugShapes proto containing the debug shapes
  *
  * Could use LOG(VISUALIZE) to plot these values. Example:
- *  LOG(VISUALIZE) << *createDebugShapesMap({
- *      {"circle_name", *createShapeProto(circle_object)},
- *      {"stadium_name", *createShapeProto(stadium_object)},
- *      {"polygon_name", *createShapeProto(polygon_object)}
+ *  LOG(VISUALIZE) << *createDebugShapes({
+ *       *createDebugShape(circle, unique_id1, optional_text),
+ *       *createDebugShape(polygon, unique_id2, optional_text),
+ *       *createDebugShape(stadium, unique_id3, optional_text)
  *  });
  *
- * @param named_shapes The map of name shape proto pairs to plot
+ * @param debug_shapes A list of debug shapes proto to plot
  *
- * @return The unique_ptr to a TbotsProto::DebugShapesMap proto containing data with
+ * @return The unique_ptr to a TbotsProto::DebugShapes proto containing data with
  *        specified names and shapes
  */
-std::unique_ptr<TbotsProto::DebugShapesMap> createDebugShapesMap(
-    const std::map<std::string, TbotsProto::Shape>& named_shapes);
+std::unique_ptr<TbotsProto::DebugShapes> createDebugShapes(
+    const std::vector<TbotsProto::DebugShapes::DebugShape>& debug_shapes);
```

Comment:
by #2901 we should be calling these functions `createDebugShapesProto` and `createDebugShapeProto`

Reply:
I think it's helpful to have `Proto` suffix in the name of conversion functions that convert a type that we have both a C++ and protobuf type of to avoid ambiguity. In this case where the type is only defined in protobuf I was thinking that it might be okay to not have the suffix in favor of a slightly shorter name to make the debug lines shorter. If you think the consistency in the name is preferred I can update it.

Reply:
in this case, I think it's okay to leave it as it is. I don't think anyone is going to be confused by it